sdp/processors/datasets/fleurs/create_initial_manifest.py

```diff
@@ -0,0 +1,160 @@
+                    - ``"train"``
+                    - ``"dev"``
+        audio_dir (str): Path to the directory where audio files will be downloaded and extracted.
+    Returns:
+        This processor generates an initial manifest file with the following fields:
+        {
+            "audio_filepath": <path to the audio file>,
+            "text": <transcription>,
+        }
+    """
+
+    def __init__(
+        self,
+        lang: str,
+        split: str,
+        raw_data_dir: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.lang = lang
+        self.split = split
+        self.raw_data_dir = raw_data_dir
+
+    def process_transcript(self, file_path: str) -> list[dict[str, typing.Any]]:
+        """
+        Parse transcript TSV file and put it inside manyfest.
+        Assumes the TSV file has two columns: file name and text.
+        """
+
+        entries = []
+        root = os.path.dirname(file_path)
+
+        with open(file_path, encoding="utf-8") as fin:
+            for line in fin:
+                # Split the line into filename text using the tab delimiter
+                parts = line.strip().split('\t')
+                if len(parts) < 2:  # Skip lines that don't have at least 2 parts
+                    continue
+
+                file_name, transcript_text = parts[1], parts[2]
+                wav_file = os.path.join(root, file_name)
+
+                entry = {"audio_filepath": os.path.abspath(wav_file), "text": transcript_text}
+                entries.append(entry)
+
+        return entries
+
+    def process_data(self, data_folder: str, manifest_file: str) -> None:
+        files = []
+        entries = []
+
+        for root, _, filenames in os.walk(data_folder):
+            for filename in fnmatch.filter(filenames, "*.tsv"):
+                files.append(os.path.join(root, filename))
+
+        for file in files:
+            result = self.process_transcript(file)
+            entries.extend(result)
+
+        with open(manifest_file, "w", encoding="utf-8") as fout:
+            for m in entries:
+                fout.write(json.dumps(m, ensure_ascii=False) + "\n")
+
+    def download_extract_files(self, dst_folder: str) -> None:
+        """downloading and extracting files"""
+
+        os.makedirs(dst_folder, exist_ok=True)
+
+        # downloading all files
+        for file_url in get_fleurs_url_list(self.lang, self.split):
+            download_file(file_url, str(dst_folder))
+        for data_file in glob.glob(f'{dst_folder}/*.tar.gz'):
+            extract_archive(str(data_file), str(dst_folder), force_extract=True)
+
+        # Organizing files into their respective folders
+        folder_mappings = {'test': '*test.tsv', 'train': '*train.tsv', 'dev': '*dev.tsv'}
+
+        for folder, file_pattern in folder_mappings.items():
+            target_folder = os.path.join(dst_folder, folder)
+
+            if not os.path.exists(target_folder):
+                continue
```

Comment:
This logic is basically to account for the fact that the user will be processing different splits?
Maybe we should just operate on only the data split folder? So for `folder` use `self.data_split` and for `file_pattern` use `f"*{self.data_split}.tsv`?